raiden/network/proxies/user_deposit.py

```diff
@@ -519,22 +526,29 @@ def _deposit_preconditions(
     @contextmanager
     def _deposit_inflight(
         self, beneficiary: Address, total_deposit: TokenAmount
-    ) -> Iterator[None]:
+    ) -> Iterator[InflightDeposit]:
         """Updates the `_inflight_deposits` dictionary to handle concurrent deposits.
 
-        Note: This must be called after `_deposit_preconditions`.
+        Note:
+          This must be called after `_deposit_preconditions`.
+          The returned InflightDeposit.async_result has to be set within the 'with' block,
+          when no Exception is raised
         """
         async_result = AsyncResult()
         current_inflight = InflightDeposit(total_deposit, async_result)
         self._inflight_deposits[beneficiary] = current_inflight
 
         try:
-            yield
+            yield current_inflight
         except Exception as e:
             async_result.set_exception(e)
             raise
         else:
-            async_result.set_result(None)
+            if not async_result.ready():
+                raise RuntimeError(
```

Comment:
I was not sure wether to raise a runtime check on coding errors or wether a 
lint rule would be more appropriate here.

Reply:
Maybe I'm missing something obvious, but why should the yield raise an exception?

Reply:
Before, this contextmanager was solely used to wake up "threads" that were waiting on the inflight deposit.
Now the successful tx-hash has to be communicated to those threads. Since the hash is only accessible inside the with block (after the yield, but before the return to the teardown actions of the contextmanager),
an `InflightDesposit.async_result.set(tx-hash)` has to be called within the with block.
This is what I want to enforce the programmer to do. 
https://github.com/ezdac/raiden/blob/772f74d2cfefcb0d6f5b3d758a7f4cb1c1270c0b/raiden/network/proxies/user_deposit.py#L326-L343

Reply:
If the only reason for the result not being ready is a programming error, then using an assert seems to be the appropriate thing.

Reply:
Implemented as assertion